design/gc-for-text-type.md

```diff
@@ -0,0 +1,37 @@
+---
+title: gc-for-text-type
+target-version: 0.1.1
+---
+
+# Garbage collection for Text Type
+
+## Summary
+
+Text and RichText use `RgaTreeSplit`. And when the text is removed, it marks a tombstone on the node corresponding to the text.  
+We provide garbage collection to clean up nodes of text type data that have been removed.
+
+### Goals
+
+When all synchronization between clients is complete, the node marked tombstone is no longer needed.
+As the number of nodes marked tombstone increases, unnecessary memory waste occurs.
+So it provides garbage collection to solve the problem of wasting memory.
```

Comment:
I think this explanation fits better in the Summary. I think it is better than what is currently described in the Summary.

In this section, it would be better to write a specific goal rather than a description of the situation.

```
Implements garbage collection for text types, Text, and RichText. When
executing garbage collection of a document, text nodes marked with
tombstones are also deleted.
```